### DISCUSSION TITLE
离线环境下如何进行数据迁移

### DISCUSSION THREAD

**Discussion by ghjhhyuyuy** [milvus-io/milvus#37858] [Q&A and General discussion]
离线环境下单机版milvus2.1的数据迁移到集群版milvus2.4，如何操作比较合适？

**Comment by yhmo** [comment]
2.1版本那时还没有备份工具，并且2.1的版本元数据格式和2.4的也不太相同，无法直接使用2.4的备份工具。
一个可能的办法是一路升级打怪到2.4，然后使用2.4的备份工具
从2.1升2.2： https://milvus.io/docs/v2.2.x/upgrade_milvus_standalone-operator.md
从2.2升2.3： https://milvus.io/docs/v2.3.x/upgrade_milvus_standalone-operator.md
从2.3升2.4： https://milvus.io/docs/upgrade_milvus_standalone-operator.md
最后再使用2.4的备份工具：https://milvus.io/docs/milvus_backup_overview.md

附带一个关于备份恢复的文档：
[milvus-backup tool practice.pdf](https://github.com/user-attachments/files/17839116/milvus-backup.tool.practice.pdf)
你直接翻到“Two Milvus in two S3”那章看就行。
